testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,5 +178,62 @@ public void testLateralViewJsonTupleAs()
                 "SELECT * FROM test_json_tuple_view",
                 queryAssert -> queryAssert.containsOnly(row(3, "Mateusz", "Gajewski", "true", "1000", null, null)));
     }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testUnionAllViews()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS union_helper");
+        onHive().executeQuery("CREATE VIEW union_helper AS\n"
+                + "SELECT (n_regionkey % 3) AS n_regionkey FROM nation");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS union_all_view");
+        onHive().executeQuery("CREATE VIEW union_all_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION ALL\n"
+                + "SELECT n_regionkey FROM union_helper\n");
```

Comment:
nit

```suggestion
        onHive().executeQuery("CREATE VIEW union_all_view AS "
                + "SELECT n_regionkey FROM nation "
                + "UNION ALL "
                + "SELECT n_regionkey FROM union_helper");
```

Reply:
I do this because it makes it easier to tell that every line in Java is separated by whitespace to avoid confusing SQL syntax errors.

(And also so it's formatted nicely in IntelliJ's "edit injected fragment" and maybe in the "convert to text block" action if we ever get to Java 15.)